Metric_Publisher/Functions/Metrics.py

```diff
@@ -0,0 +1,56 @@
+from Metric_Publisher.Functions.Metric import Metric
+from contextlib import contextmanager
+import time
+
+
+class TimingMetric(Metric):
+    """Timing metric that calculates the time the metric has run."""
+
+    def __init__(self, metric_name: str, unit: str):
+        super().__init__(metric_name, unit)
+
+    @contextmanager
+    def timeit_metric(self):
+        """Context manager used to calculate the time the method has run."""
+        start_time = 0
+        try:
+            start_time = time.time()
+            yield
+        finally:
+            self._value = time.time() - start_time
+
+    def get_name(self) -> str:
+        """Returns the metric name."""
+        return super().get_name()
+
+    def get_unit(self) -> str:
+        """Returns the unit associated with the metric."""
+        return super().get_unit()
+
+    def get_value(self):
+        """Returns the value associated with the metric."""
+        return super().get_value()
```

Comment:
If it's just calling super, no need to define the method in the subclass. Just leave it blank